### PR TITLE
DNS default check should not be in server.Config

### PR DESCRIPTION
**Pull request by smarterclayton**
Prints excess messages anytime the CLI is loaded

@deads2k review

### CODE REVIEW

File: pkg/cmd/server/origin_master.go [hunk 1]
```diff
@@ -83,10 +85,18 @@ func (cfg Config) BuildOriginMasterConfig() (*origin.MasterConfig, error) {
 		return nil, err
 	}
 
+	dnsAddr := cfg.DNSBindAddr
```
Comment (deads2k) [inline pkg/cmd/server/origin_master.go]: This validation should happen in a separate method call for GetDNSAddr like we do for other addresses that require detection.  https://github.com/openshift/origin/blob/master/pkg/cmd/server/config.go#L123 as an example.

**Comment by deads2k** [comment]
One comment, but this is getting nuked one more time to properly delegate commands in the order we want with reusable code and eliminating at least one translation layer in #1247, so merging as-is would be ok.

**Comment by deads2k** [comment]
@csrwng

File: pkg/cmd/server/origin_master.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/origin_master.go]: It's not appropriate for it to be called more than once, so I don't want to encourage reuse.

Comment (liggitt) [inline pkg/cmd/server/origin_master.go]: this feels similar to the warnings you get when :80 and :443 can't be bound to, but those happen at start time. I would move this earlier (to a validate func) or later (to the start func). Doing it in buildconfig feels weird

Comment (smarterclayton) [inline pkg/cmd/server/origin_master.go]: Fixed

----- Original Message -----

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**